app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/IvController.scala

```diff
@@ -59,9 +75,64 @@ class IvController @Inject()(
     }
   }
 
-  def ivFailure(journeyId: UUID): Action[AnyContent] = Action { implicit request =>
-    logger.warn(s"IV journey failed: ${journeyId.toString}")
-    InternalServerError(errorHandler.internalServerErrorTemplate)
+  def retry():  Action[AnyContent] = authenticatedActionWithSessionData.async {
+    implicit request => redirectToIv
+  }
+
+  def ivFailureCallback(journeyId: UUID): Action[AnyContent] = authenticatedActionWithSessionData.async {
+    implicit request =>
+      ivService
+        .getFailedJourneyStatus(journeyId)
+        .fold(
+          { e =>
+            logger.warn("Could not check IV journey error status", e)
+            Redirect(routes.IvController.getTechnicalIssue())
+          }, {
+            case IvErrorStatus.Incomplete           => Redirect(routes.IvController.getTechnicalIssue())
+            case IvErrorStatus.FailedMatching       => Redirect(routes.IvController.getFailedMatching())
+            case IvErrorStatus.FailedIV             => Redirect(routes.IvController.getFailedIV())
+            case IvErrorStatus.InsufficientEvidence => Redirect(routes.IvController.getInsufficientEvidence())
+            case IvErrorStatus.LockedOut            => Redirect(routes.IvController.getLockedOut())
+            case IvErrorStatus.UserAborted          => Redirect(routes.IvController.getUserAborted())
+            case IvErrorStatus.Timeout              => Redirect(routes.IvController.getTimedOut())
+            case IvErrorStatus.TechnicalIssue       => Redirect(routes.IvController.getTechnicalIssue())
+            case IvErrorStatus.PreconditionFailed   => Redirect(routes.IvController.getPreconditionFailed())
+            case IvErrorStatus.Unknown(value) =>
+              logger.warn(s"Received unknown error response status from IV: $value")
```

Comment:
do we not need the FailedDirectorCheck as mentioned in the design slides?

Reply:
when we talked to the auth team, they said this was a status which existed but shouldn't as they never return it and are going to take it out. As the code is written it would just get swallowed up as "Unknown".

Reply:
Looking at the `identity-verification-frontend` service README, we may not need to do this as this would mean verifying director info with company house. I don't think we have that as a business case